runtime/lua/vim/lsp.lua

```diff
@@ -855,7 +855,8 @@ do
       local lines = nvim_buf_get_lines(bufnr, 0, -1, true)
       local startline =  math.min(firstline + 1, math.min(#client._cached_buffers[bufnr], #lines))
       local endline =  math.min(-(#lines - new_lastline), -1)
-      local incremental_change = vim.lsp.util.compute_diff(client._cached_buffers[bufnr], lines, startline, endline)
+      local incremental_change = vim.lsp.util.compute_diff(
+        client._cached_buffers[bufnr], lines, startline, endline, client.offset_encoding or "utf-16")
```

Comment:
I think we need to remove the `once` call further above now,  to make sure this works with a mix of utf-8/utf-16 clients. 
Otherwise it re-uses the result of whichever client comes first. 

It was already wonky to depend on the cached buffers being the same for all clients

Reply:
Addressed. Do you think this is good to merge now?

Reply:
yep :+1: